src/nagini_translation/translators/call.py

```diff
@@ -149,6 +149,35 @@ def _translate_super(self, node: ast.Call, ctx: Context) -> StmtsAndExpr:
         else:
             raise InvalidProgramException(node, 'invalid.super.call')
 
+    def translate_adt_cons(self, cons: PythonClass, args: List, pos,
+                           ctx: Context) -> Expr:
+        """
+        Constructs ADTs via a sequence of constructor calls and
+        boxing/unboxing calls.
+        """
+        info = self.no_info(ctx)
+        adt_name = cons.adt_def.name + '_ADT'
```

Comment:
This looks like it doesn't use the fresh_name mechanism? But it probably should?

Reply:
Ok, I updated now using the fresh name mechanism. I'm using the class scope, I hope this is enough.